src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVCleanPt1a.java

```diff
@@ -0,0 +1,439 @@
+                    (variant.getEnd() - variant.getStart() >= MIN_ALLOSOME_EVENT_SIZE)) {
+                final String sampleName = genotype.getSampleName();
+                final int sex = sampleSexMap.get(sampleName);
+                if (sex == 1 && isRevisableEvent(variant, isY)) { // Male
+                    writeRevisedEvents(variant);
+                    adjustMaleGenotype(genotype, genotypeBuilder, svType);
+                } else if (sex == 2 && isY) { // Female
+                    genotypeBuilder.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+                } else if (sex == 0) { // Unknown
+                    genotypeBuilder.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+                }
+            }
+        }
+    }
+
+    private void adjustMaleGenotype(Genotype genotype, GenotypeBuilder genotypeBuilder, String svType) {
+        if (genotype.hasExtendedAttribute(GATKSVVCFConstants.RD_CN)) {
+            final int rdCN = Integer.parseInt(genotype.getExtendedAttribute(GATKSVVCFConstants.RD_CN).toString());
+            genotypeBuilder.attribute(GATKSVVCFConstants.RD_CN, rdCN + 1);
+
+            final Allele refAllele = genotype.getAllele(0);
+            final Allele altAllele = genotype.getAllele(1);
+            if (svType.equals(GATKSVVCFConstants.SYMB_ALT_STRING_DEL)) {
+                if (rdCN >= 1) genotypeBuilder.alleles(Arrays.asList(refAllele, refAllele));
+                else if (rdCN == 0) genotypeBuilder.alleles(Arrays.asList(refAllele, altAllele));
+            } else if (svType.equals(GATKSVVCFConstants.SYMB_ALT_STRING_DUP)) {
+                if (rdCN <= 1) genotypeBuilder.alleles(Arrays.asList(refAllele, refAllele));
+                else if (rdCN == 2) genotypeBuilder.alleles(Arrays.asList(refAllele, altAllele));
+                else genotypeBuilder.alleles(Arrays.asList(altAllele, altAllele));
+            }
+        }
+    }
+
+    private boolean isRevisableEvent(VariantContext variant, boolean isY) {
+        final List<Genotype> genotypes = variant.getGenotypes();
+        int[] maleCounts = new int[4];
+        int[] femaleCounts = new int[4];
+        for (final Genotype genotype : genotypes) {
+            final String sampleName = genotype.getSampleName();
+            final Integer sex = sampleSexMap.get(sampleName);
+            if (sex == null) continue;
+
+            int rdCN = (int) genotype.getExtendedAttribute(GATKSVVCFConstants.RD_CN, -1);
+            if (rdCN == -1) continue;
+
+            final int rdCNVal = Math.min(rdCN, 3);
+            if (sex == 1) {
+                maleCounts[rdCNVal]++;
+            } else if (sex == 2) {
+                femaleCounts[rdCNVal]++;
+            }
+        }
+
+        final double maleMedian = calcMedian(maleCounts);
+        final double femaleMedian = calcMedian(femaleCounts);
+        return maleMedian == 1.0 && (isY ? femaleMedian == 0.0 : femaleMedian == 2.0);
```

Comment:
I would always avoid using `==` with floating-point types because of precision errors. Can you change this to int types?